Cabal/Distribution/Simple/Utils.hs

```diff
@@ -829,53 +828,52 @@ rawSystemStdInOut verbosity path args mcwd menv input outputMode = withFrozenCal
       -- fork off a couple threads to pull on the stderr and stdout
       -- so if the process writes to stderr we do not block.
 
-      err <- hGetContents errh
-
-      out <- IOData.hGetContents outh outputMode
-
-      mv <- newEmptyMVar
-      let force str = do
-            mberr <- Exception.try (evaluate (rnf str) >> return ())
-            putMVar mv (mberr :: Either IOError ())
-      _ <- forkIO $ force out
-      _ <- forkIO $ force err
-
-      -- push all the input, if any
-      case input of
-        Nothing -> return ()
-        Just inputData -> do
-          -- input mode depends on what the caller wants
-          IOData.hPutContents inh inputData
-          --TODO: this probably fails if the process refuses to consume
-          -- or if it closes stdin (eg if it exits)
-
-      -- wait for both to finish, in either order
-      mberr1 <- takeMVar mv
-      mberr2 <- takeMVar mv
-
-      -- wait for the program to terminate
-      exitcode <- waitForProcess pid
-      unless (exitcode == ExitSuccess) $
-        debug verbosity $ path ++ " returned " ++ show exitcode
-                       ++ if null err then "" else
-                          " with error message:\n" ++ err
-                       ++ case input of
-                            Nothing       -> ""
-                            Just d | IOData.null d -> ""
-                            Just (IODataText inp) -> "\nstdin input:\n" ++ inp
-                            Just (IODataBinary inp) -> "\nstdin input (binary):\n" ++ show inp
-
-      -- Check if we we hit an exception while consuming the output
-      -- (e.g. a text decoding error)
-      reportOutputIOError mberr1
-      reportOutputIOError mberr2
-
-      return (out, err, exitcode)
+      let force :: NFData a => a -> IO a
+          force str = do
+            evaluate (rnf str)
```

Comment:
it's a bit confusing as this clashes with `Control.DeepSeq.force` (which is available since deepseq-1.2; and Cabal requires deepseq>=1.3); if we're touching this code anyway, why not simply import `force` and use `(evaluate . force)` below? or at least call this local helper `evaluateNF` or `forceIO` instead :-)

alternatively, if this is a common pattern (it's quite common you'd want the backgrounded task to do the actual computation than to end up with some lazy thunk returned rightaway which would render the backgrounding pointless...) you could define a variant `withAsyncNF` which encodes this common use-pattern

Reply:
The name `force` was there already, and I didn't tried too hard to come with new one.

`withAsyncNF` is a good point, but won't do it for this sole use case.